docs/user/templates.rst

```diff
@@ -71,44 +74,84 @@ here any files that you don't want to be tracked.
 
 This yml file is for "continuous integration" in GitLab. It is configured so
 that every time you make a commit, your project's docker image is rebuilt. In
-most cases, this build should be successful. If, however, you are making modifications
-to the ``Dockerfile``, you should pay attention to the CI/CD tab in GitLab to check
-for failing builds. Take a look at :ref:`advanced_interfaces`.
+most cases, this build should be successful. If, however, you are making
+modifications to the ``Dockerfile``, you should pay attention to the CI/CD tab
+in GitLab to check for failing builds. Take a look
+at :ref:`advanced_interfaces`.
 
 ``Dockerfile``
 """"""""""""""
 
 When you run the notebook server, a Docker image is built for your project as
-defined by this ``Dockerfile``. The FROM line in this ``Dockerfile`` defines which
-Renku Docker image sets up the base of your project; this includes dependencies
-for the Renku CLI, jupyterlab, and maybe R kernels & RStudio, depending on which
-template you selected upon project creation.
+defined by this ``Dockerfile``. The FROM line in this ``Dockerfile`` defines
+which Renku Docker image sets up the base of your project; this includes
+dependencies for the Renku CLI, Jupyterlab, and maybe R kernels & RStudio,
+depending on which template you selected upon project creation.
 
-The lines following ``FROM`` define the installation of your own software dependencies;
-they are the instructions for conda/pip installations of the libraries in your
-``requirements.txt`` file. If there's nothing special about the libraries you're
-installing, you wont have to make changes to this file. Else, check
-out :ref:`advanced_interfaces`.
+The lines following ``FROM`` define the installation of your own software
+dependencies; they are the instructions for conda/pip installations of the
+libraries in your ``requirements.txt`` file. If there's nothing special
+about the libraries you're installing, you wont have to make changes to this
+file. Else, check out :ref:`advanced_interfaces`.
 
 ``.dockerignore``
 """""""""""""""""
 
-The ``.dockerignore`` file is just like a ``.gitignore`` file, in that it allows
-you to specify which files to ignore in a docker build. If you aren't making
-changes to the docker build, you can .dockerignore this file.
+The ``.dockerignore`` file is just like a ``.gitignore`` file, in that it
+allows you to specify which files to ignore in a docker build. If you aren't
+making changes to the docker build, you can .dockerignore this file.
 
 ``README.md``
 """""""""""""
 
-The ``README.md`` file is shown on a project's home page. It's good to have at least
-the name of the project and a brief overview of the project for your intended
-audience.
+The ``README.md`` file is shown on a project's home page. It's good to have at
```

Comment:
Proposed change to this section:

```
``.renku``
""""""""""

The ``.renku`` directory includes a ``renku.ini`` file which contains 
project-level configuration for renku, stored using the 
`INI format <https://en.wikipedia.org/wiki/INI_file>`_. Currently, it 
can be used to specify defaults values for launching interactive environments.

**Interactive Environments**

If your project has specific resources requirements to run, or if it should default
to RStudio or anything other than JupyterLab, then you will want to provide a
configuration for the interactive environments.

Although the file may be modified manually, it is recommended to use the
``renku config --local interactive.<property> <value>`` command.

Here is the list of properties that can be customized in a standard Renkulab
deployment:

* ``default_url [string]``: url to use when starting a new interactive
  environment
* ``cpu_request [float]``: CPUs quota
* ``mem_request [float]``: memory quota (in GBs)
* ``gpu_request [int]``: GPU quota
* ``lfs_auto_fetch [bool]``: whether to automatically fetch lfs files or not

.. note::

    We use JupyterLab as the default web interface for interactive environments. If you work in R, you may prefer to
    have RStudio ad the default. This can be achieved by using `/rstudio` as the ``default_url`` instead of `/lab`.

    .. code-block:: console

      > renku config --local interactive.default_url "/rstudio"

    Verify that your renku.ini file looks like the following.

    .. code-block:: console

      [renku "interactive"]
      default_url = /rstudio

    If you ran this command locally, you will need push back to the renkulab server, e.g., 
    
    .. code-block:: console

      > git push
    
    before this change is available. (`renku config` automatically creates a commit).

    You can now start a new environment against the latest commit and you will have RStudio as the
    default web interface.
    

.. note::

    Using the same approach as above for RStudio, it is possible to switch the interface from JupyterLab to
    the classic Jupyter Notebook by using `/tree` as the ``default_url`` instead of `/lab`.

    .. code-block:: console

      > renku config --local interactive.default_url "/tree"
```